app/routing/ArrivalsRouter.scala

```diff
@@ -42,28 +44,34 @@ class ArrivalsRouter @Inject() (
     with StreamingParsers
     with VersionedRouting {
 
-  def injectEISResponse(arrivalId: String): Action[Source[ByteString, ?]] = route {
-    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE) =>
-      (for {
-        convertedArrivalId <- TransitionalBindings.movementIdBinding.bind("arrivalId", arrivalId)
-      } yield convertedArrivalId).fold(
-        bindingFailureAction(_),
-        convertedArrivalId => v2Arrivals.sendArrivalsResponse(convertedArrivalId)
-      )
-    case Some(VersionedRouting.VERSION_2_1_ACCEPT_HEADER_VALUE) =>
-      (for {
-        convertedArrivalId <- Bindings.movementIdBinding.bind("arrivalId", arrivalId)
-      } yield convertedArrivalId).fold(
-        bindingFailureAction(_),
-        convertedArrivalId => arrivals.sendArrivalsResponse(convertedArrivalId)
-      )
-    case _ =>
-      (for {
-        convertedArrivalId <- implicitly[PathBindable[ArrivalId]].bind("arrivalId", arrivalId)
-      } yield convertedArrivalId).fold(
-        bindingFailureAction(_),
-        convertedArrivalId => v1Arrivals.injectEISResponse(convertedArrivalId)
-      )
+  def injectEISResponseWithTriggerId(arrivalId: String, messageId: String): Action[Source[ByteString, _]] = routing(arrivalId, messageId.some)
+  def injectEISResponse(arrivalId: String): Action[Source[ByteString, ?]]                                 = routing(arrivalId = arrivalId, messageId = None)
```

Comment:
Test support is written to replicate ERMIS and EIS in local. So here with this we always assume triggerid. It is not going to be optional in test support. I suggest change the existing endpoints by adding new mandatory messageId(triggerId).

Reply:
This call is not from common-transit-convention-traders so triggerId should be present(Part of conversationId last 16 digit)